public/app/features/explore/TraceView/components/TracePageHeader/TracePageHeader.tsx

```diff
@@ -110,76 +110,184 @@ export const TracePageHeader = memo((props: TracePageHeaderProps) => {
-              </span>
-            </Tooltip>
-          )}
-          {status && status.length > 0 && (
-            <Tooltip
-              // eslint-disable-next-line @grafana/i18n/no-untranslated-strings
-              content="http.status_code"
-              interactive={true}
-            >
-              <span className={styles.tag}>
-                <Badge text={status[0].value} color={statusColor} />
-              </span>
-            </Tooltip>
-          )}
-          {url && url.length > 0 && (
-            <Tooltip content={urlTooltip(url[0].value)} interactive={true}>
-              <span className={styles.url}>{url[0].value}</span>
+        {/* Action buttons */}
+        <div className={styles.actions}>
+          <div className={styles.shareButtonGroup}>
+            <Tooltip content={t('explore.trace-page-header.share-tooltip', 'Share trace')}>
+              <Button
+                size="sm"
+                variant="secondary"
+                fill="outline"
+                icon="share-alt"
+                onClick={() => {
+                  navigator.clipboard.writeText(window.location.href);
+                  notifyApp.success(t('explore.trace-page-header.link-copied', 'Link copied to clipboard'));
+                }}
+              >
+                {t('explore.trace-page-header.share', 'Share')}
+              </Button>
             </Tooltip>
+
+            <Dropdown overlay={shareDropdownMenu} placement="bottom-end">
+              <Button
+                size="sm"
+                variant="secondary"
+                fill="outline"
+                icon="angle-down"
+                className={styles.shareDropdownButton}
+              />
+            </Dropdown>
+          </div>
+
+          {/* Plugin extension actions */}
+          {extensionLinks.length > 0 && (
+            <div className={styles.actions}>
+              {extensionLinks.map((link) => (
+                <Tooltip key={link.id} content={link.description || link.title}>
+                  <Button
+                    size="sm"
+                    variant="secondary"
+                    fill="outline"
+                    icon={link.icon}
+                    onClick={(event) => {
+                      if (link.path) {
+                        window.open(link.path, '_blank');
+                      }
+                      link.onClick?.(event);
+                    }}
+                  >
+                    {link.title}
+                  </Button>
+                </Tooltip>
+              ))}
+            </div>
           )}
-        </span>
+        </div>
+      </div>
+
+      {/* Metadata row */}
+      <div className={styles.metadataRow}>
+        <div className={styles.metadataItem}>
+          <span className={styles.metadataLabel}>{t('explore.trace-page-header.trace-id', 'Trace ID')}</span>
+          <span className={styles.metadataValue}>
+            <button className={styles.traceIdButton} onClick={copyTraceId}>
```

Comment:
Nice little trick with icon switching to "check" after copying. It'd be nice to have something like this in grafana/ui